python/pyiceberg/schema.py

```diff
@@ -201,7 +203,7 @@ def find_type(self, name_or_id: Union[str, int], case_sensitive: bool = True) ->
 
     @property
     def highest_field_id(self) -> int:
-        return visit(self.as_struct(), _FindLastFieldId())
+        return max(self._lazy_id_to_name.keys(), default=0)
```

Comment:
This seems unrelated, though fine.

Reply:
Should have been a separate PR indeed, but it is a very welcome change

Reply:
The old way didn't get the highest field id correctly, so I changed it.